### DISCUSSION TITLE
Confining Telegram installed in user home w/ firejail-0.9.66

### DISCUSSION THREAD

**Discussion by domivogt** [netblue30/firejail#4739] [Q&A] [answered]
Debian based System with firejail-0.9.66; Telegram installed in ~/Telegram.  What's the problem, please?

```
 $ firejail --debug --profile=/etc/firejail/Telegram.profile ~/Telegram/Telegram
...
Warning: Cannot confine the application using AppArmor.
Maybe firejail-default AppArmor profile is not loaded into the kernel.
As root, run "aa-enforce firejail-default" to load it.
Starting application
LD_PRELOAD=(null)
execvp argument 0: /home/<username>/Telegram/Telegram
Child process initialized in 190.81 ms
Error: no suitable /home/<username>/Telegram/Telegram executable found  <------ ?????
monitoring pid 23

Sandbox monitor: waitpid 23 retval 23 status 256

Parent is shutting down, bye...

```

There is no ~/.TelegramDesktop directory (which is mentioned in the profile).

**Comment by kmk3** [comment]
@domivogt [on Dec 3](https://github.com/netblue30/firejail/discussions/4739#discussion-3721757):

> Debian based System with firejail-0.9.66; Telegram installed in ~/Telegram.
> What's the problem, please?
> 
> ```
>  $ firejail --debug --profile=/etc/firejail/Telegram.profile ~/Telegram/Telegram
> ...
> Warning: Cannot confine the application using AppArmor.
> Maybe firejail-default AppArmor profile is not loaded into the kernel.
> As root, run "aa-enforce firejail-default" to load it.
> Starting application
> LD_PRELOAD=(null)
> execvp argument 0: /home/<username>/Telegram/Telegram
> Child process initialized in 190.81 ms
> Error: no suitable /home/<username>/Telegram/Telegram executable found  <------ ?????
> monitoring pid 23
> 
> Sandbox monitor: waitpid 23 retval 23 status 256
> 
> Parent is shutting down, bye...
> ```

> `Error: no suitable /home/<username>/Telegram/Telegram executable found  <------ ?????`

telegram.profile has `include disable-exec.inc`, which has `noexec ${HOME}`. So
try this (on e.g.: ~/.config/firejail/telegram.local):

```firejail
ignore noexec ${HOME}
```

> There is no ~/.TelegramDesktop directory (which is mentioned in the profile).

If you are going to put all the data on ~/Telegram as well, you can probably
ignore the other directories:

```firejail
ignore noblacklist ${HOME}/.TelegramDesktop
ignore noblacklist ${HOME}/.local/share/TelegramDesktop

ignore mkdir ${HOME}/.TelegramDesktop
ignore mkdir ${HOME}/.local/share/TelegramDesktop
ignore whitelist ${HOME}/.TelegramDesktop
ignore whitelist ${HOME}/.local/share/TelegramDesktop
```

**Comment by domivogt** [comment]
That doesn't seem to help.  Even after removing disable-exec.inc from the system wide profile:

```
$ cat ~/.config/firejail/telegram.local  
ignore noexec ${HOME}
$ firejail ~/Telegram/Telegram
Reading profile /etc/firejail/Telegram.profile
Reading profile /etc/firejail/telegram.profile
Reading profile /home/<username>n/.config/firejail/telegram.local
Reading profile /etc/firejail/disable-common.inc
Reading profile /etc/firejail/disable-devel.inc
Reading profile /etc/firejail/disable-interpreters.inc
Reading profile /etc/firejail/disable-passwdmgr.inc
Reading profile /etc/firejail/disable-programs.inc
Reading profile /etc/firejail/disable-shell.inc
Reading profile /etc/firejail/disable-xdg.inc
...
Error: no suitable /home/luthien/Telegram/Telegram executable found
```

**Reply by kmk3** [reply]
@domivogt [on Dec 3](https://github.com/netblue30/firejail/discussions/4739#discussioncomment-1746365):

> That doesn't seem to help. Even after removing disable-exec.inc from the
> system wide profile:

Is ~/Telegram/Telegram a shell script?  If so, try adding this:

```firejail
include allow-bin-sh.inc

private-bin bash,dash,sh
```

**Comment by rusty-snake** [comment, marked as answer]
Try this telegram.local:

```
ignore noexec ${HOME}
ignore apparmor
whitelist ${HOME}/Telegram
```

**Comment by domivogt** [comment]
Jep, thanks, that works.
